### PR TITLE
Jormungandr: Condition to stop Kraken calls

### PR DESCRIPTION
**Pull request by mbounabi**
https://jira.kisio.org/browse/NAVP-904

### CODE REVIEW

**Comment by nonifier** [comment]
much tidier !

**Review by nonifier — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 1]
```diff
@@ -883,22 +888,28 @@ def fill_journeys(self, request_type, api_request, instance):
...
-                journey_filter.filter_shared_sections_journeys(pool2, api_request)
+            journey_filter.filter_similar_vj_journeys(journeys_pool, api_request)
 
             responses.extend(new_resp)  # we keep the error for building the response
 
             nb_qualified_journeys = nb_journeys(responses)
 
-            # We allow one more call to kraken if there is no valid journey.
+            if nb_previously_qualified_journeys == nb_qualified_journeys:
+                # If there are no qualified journey in the kraken response,
```
Comment (kadhikari) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: If there is no additional qualified journey in the last kraken response

**Review by kadhikari — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 2]
```diff
@@ -823,21 +823,26 @@ def fill_journeys(self, request_type, api_request, instance):
...
 
-        min_journeys_calls = get_or_default(request, '_min_journeys_calls', 1)
-
         responses = []
         nb_try = 0
         nb_qualified_journeys = 0
+        nb_previously_qualified_journeys = 0
+        last_chance_retry = False
 
+        min_journeys_calls = get_or_default(request, '_min_journeys_calls', 1)
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: `request.get('_min_journeys_calls', 1)` doesn't work?

**Review by kinnou02 — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 2 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/scenarios/new_default.py]: Not tested. This line was just moved here, it isn't related to this PR.

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 3]
```diff
@@ -883,22 +888,28 @@ def fill_journeys(self, request_type, api_request, instance):
...
+            nb_previously_qualified_journeys = nb_qualified_journeys
+
             if nb_qualified_journeys == 0:
                 min_journeys_calls = max(min_journeys_calls, 2)
-
+                
         logger.debug('nb of call kraken: %i', nb_try)
 
+        if api_request['no_shared_section']:
+            journey_filter.filter_shared_sections_journeys(journey_filter.get_qualified_journeys(responses), api_request)
```
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/new_default.py:911]: depends if you are in debug mode, should probably be handled in final filter

Comment (mbounabi) [reply source/jormungandr/jormungandr/scenarios/new_default.py:911]: The final filter has its purpose : to delete journeys with the same lines sequence. So this filter has nothing to do with it. I don't see the point.
Also, the journeys filtered here will be handled as every other journeys in every other filters, (mark_as_dead in the function '_filter_similar_journeys')

**Pull request merged**